lib/spack/spack/build_environment.py

```diff
@@ -150,14 +150,19 @@ def __call__(self, *args, **kwargs):
         parallel = should_set_parallel_jobs(jobserver_support=True) and kwargs.pop(
             "parallel", self.jobs > 1
         )
+        if not parallel:
+            # Force -j1 to disable parallel even under jobserver
+            jobs = 1
```

Comment:
this branch is now followed when job server support is detected, which is incorrect.

Reply:
Yes, the first version did, but then the test failed, and I did a mistake in the new version. I'll mark this MR as draft until I find a proper fix.